tests/mock/agents.py

```diff
@@ -160,11 +160,12 @@ def assert_no_transactions(self) -> None:
         unexpected_transactions = self.get_unexpected_transactions(allowed=None)
         assert not bool(unexpected_transactions)
 
-    def reset(self, clear_side_effects: bool = True) -> None:
+    def reset(self, clear_side_effects: bool = True, clear_return_values: bool = True) -> None:
         for mock in self._MOCK_METHODS:
-            mock.reset_mock()
-            if clear_side_effects:
-                mock.side_effect = None
+            mock.reset_mock(return_value=clear_return_values, side_effect=clear_side_effects)
```

Comment:
Just to make sure: `return_value` will be same as set in `MockContractAgent`?

Reply:
No, that line alone it's not enough. That's why the next lines are needed to restore the default return values.

Reply:
Oh should I do that conditional to `clear_return_values`?

Reply:
Good point - I think so

Reply:
✔️